src/OpenTelemetry.AutoInstrumentation.Native/continuous_profiler.cpp

```
@@ -720,6 +896,36 @@ void SleepMillis(unsigned int millis)
 #endif
 }
 
+unsigned int GetSleepTime(const ContinuousProfiler* const prof)
+{
+    // Assumption is continuous profiling interval is bigger and multiple of selective sampling interval.
+    // If both are enabled, we need to wake every smaller interval.
+    if (prof->selectedThreadsSamplingInterval.has_value())
+    {
+        return prof->selectedThreadsSamplingInterval.value();
+    }
+    if (prof->threadSamplingInterval.has_value())
+    {
+        return prof->threadSamplingInterval.value();
+    }
+    // Shouldn't ever happen.
+    return 0;
+}
```

The comment assumes the continuous interval is larger and a multiple of the selective one. Would it make sense to return the smaller of the two intervals instead, to avoid relying on this assumption?

Also, instead of returning 0 in the “Shouldn’t ever happen” case, maybe consider throwing an exception, adding an assert, or using `std::unreachable()` to catch unexpected states earlier. What do you think?